### PR TITLE
Document the cobuild feature

### PR DESCRIPTION
**Pull request by octogonz**
**TODO:**

- [x] Convert Cheng's notes to markdown
- [x] Go through the steps myself to test the feature
- [x] Update `https://developer.microsoft.com/json-schemas` with `cobuild.schema.json`

@chengcyber @MichaelSitter

### CODE REVIEW

File: websites/rushjs.io/docs/pages/configs/environment_vars.md [hunk 1]
```diff
@@ -64,6 +64,28 @@ Overrides the value of `isCacheWriteAllowed` in the `build-cache.json` configura
 environment variable must be `1` (for true) or `0` (for false). If there is no build cache configured, then
 this environment variable is ignored.
 
+## RUSH_COBUILD_CONTEXT_ID (EXPERIMENTAL)
+
+Cobuild pipelines must define this environment variable; without it, Rush will perform a regular build without
+any cobuild logic. See the [Cobuilds](../maintainer/cobuilds.md) documentation for details.
+
+## RUSH_COBUILD_LEAF_PROJECT_LOG_ONLY_ALLOWED
```
Comment (chengcyber) [inline websites/rushjs.io/docs/pages/configs/environment_vars.md:72]: It seems to me that this is an EXPERIMENTAL env var as well

File: websites/rushjs.io/docs/pages/maintainer/cobuilds.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
+---
+title: Cobuilds (Experimental)
+---
+
+<!--
+  Grammar note for writers
```
Comment (chengcyber) [inline websites/rushjs.io/docs/pages/maintainer/cobuilds.md:6]: 👍

**Comment by octogonz** [comment]
JSON schemas are published: https://github.com/microsoft/json-schemas/pull/153

File: websites/rushjs.io/docs/pages/configs/environment_vars.md (hunk 1 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/configs/environment_vars.md:72]: Fixed

**Pull request merged**